Source/JSONAPISerializer.swift

```diff
@@ -72,22 +72,27 @@ public protocol JSONAPIEntity: CustomSerializable, JSONAPISerializable {
  */
 public struct JSONAPISerializer<T: JSONAPIEntity>: Serializable {
     
-    // Top level `data` member: the document’s “primary data”
-    private let data: AnyObject
+    private typealias JSONAPIConvertible = protocol<JSONAPISerializable, Serializable>
+
+    /// Top level [`data`](http://jsonapi.org/format/#document-top-level) member: the document’s “primary data”
+    private let data: JSONAPIConvertible
 
-    // Top level `included` member: an array of resource objects that are related to the primary data and/or each other (“included resources”).
+    /// Top level `included` member: an array of resource objects that are related to the primary data and/or each other (“included resources”).
     private let included: [AnyObject]?
 
-    // Top level `links` member: a links object related to the primary data.
-    private let links: AnyObject?
-    
-    private typealias JSONAPIConvertible = protocol<JSONAPISerializable, Serializable>
-    private typealias JSONAPISerializerInit = (data: AnyObject, links: AnyObject?, included: [AnyObject]?)
+    /// Top level [`links`](http://jsonapi.org/format/#document-links) member: a links object related to the primary data.
+    private let links: [String: JSONAPILink]?
+
+    /// Top level [`meta`](http://jsonapi.org/format/#document-meta) member: used to include non-standard meta-information.
+    private let meta: Serializable?
+
+    private typealias JSONAPISerializerInit = (data: JSONAPIConvertible, links: [String: JSONAPILink]?, meta: Serializable?, included: [AnyObject]?)
     
-    private static func commonInit(object: JSONAPIConvertible, topLevelLinks: [String: JSONAPILink]?, includeChildren: Bool) -> JSONAPISerializerInit {
+    private static func commonInit(object: JSONAPIConvertible, topLevelLinks: [String: JSONAPILink]?, meta: Serializable?, includeChildren: Bool) -> JSONAPISerializerInit {
```

Comment:
Also here I'd change either `topLevelLinks` to `links` or `meta` to `topLevelMeta`